packages/graph-engine/src/nodes/color/contrasting.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Performs a contrast calculation between two colors using APCA-W3 calcs
+ *
+ * @packageDocumentation
+ */
+
+import { NodeDefinition, NodeTypes } from "#/types.js";
+import { calcAPCA } from "apca-w3";
+import chroma from "chroma-js";
+
+export const type = NodeTypes.CONTRASTING;
+
+export type State = {
+    a: string,
+    b: string,
+    background: string,
+    wcag: number,
+    threshold: number,
+    contrast: number
+}
+
+type contrastingValues = {
+    color: string,
+    sufficient: boolean,
+    contrast: number
+}
+
+export const defaults: State = {
+    a: "#000000",
+    b: "#ffffff",
+    background: "#ffffff",
+    wcag: 3,
+    threshold: 60,
+    contrast: 0
+}
+
```

Comment:
We should probably add a validateInput function here to make sure someone passes through an acceptable value of the wcag value